game/campaignloader/defaultsquadronassigner.py

```diff
@@ -48,24 +49,41 @@ def assign(self) -> None:
     def find_squadron_for(
         self, config: SquadronConfig, control_point: ControlPoint
     ) -> Optional[SquadronDef]:
+        squadron_def = None
         for preferred_aircraft in config.aircraft:
             squadron_def = self.find_preferred_squadron(
                 preferred_aircraft, config.primary, control_point
             )
             if squadron_def is not None:
-                return squadron_def
+                break
 
         # If we didn't find any of the preferred types we should use any squadron
         # compatible with the primary task.
-        squadron_def = self.find_squadron_for_task(config.primary, control_point)
-        if squadron_def is not None:
-            return squadron_def
+        if squadron_def is None:
+            squadron_def = self.find_squadron_for_task(config.primary, control_point)
 
         # If we can't find any squadron matching the requirement, we should
         # create one.
-        return self.air_wing.squadron_def_generator.generate_for_task(
-            config.primary, control_point
-        )
+        if squadron_def is None:
+            squadron_def = self.air_wing.squadron_def_generator.generate_for_task(
+                config.primary, control_point
+            )
+
+        # Override squadron def with squadron config parameters from campaign file, if defined
+        if squadron_def is not None:
+
+            overrides: Dict[str, Union[str, int]] = {}
+            if config.name is not None:
+                overrides["name"] = config.name
+            if config.nickname is not None:
+                overrides["nickname"] = config.nickname
+            if config.female_pilot_ratio is not None:
+                overrides["female_pilot_ratio"] = config.female_pilot_ratio
+
+            squadron_copy = dataclasses.replace(squadron_def, **overrides)
+            return squadron_copy
+        else:
+            return None
```

Comment:
This control flow in this function cleans up well if this block gets factored out into its own function that gets called from `DefaultSquadronAssigner.assign`. The rest of the changes in this function can just be reverted, but the callsite turns into `self.override_squadron_defaults(self.find_squadron_for(...))`.

Reply:
Thanks.
I applied your suggestion in 61c1d12a86bf1e21aa3470a2996ea23b9052096f